impl/src/main/java/org/ehcache/config/builders/ConfigurationBuilder.java

```diff
@@ -78,7 +78,9 @@ public Configuration build() {
 
   ConfigurationBuilder addCache(String alias, CacheConfiguration<?, ?> config) {
     Map<String, CacheConfiguration<?, ?>> newCaches = new HashMap<String, CacheConfiguration<?, ?>>(caches);
-    newCaches.put(alias, config);
+    if(newCaches.put(alias, config) != null) {
```

Comment:
With our immutable builders, we promote sharing them. Should we have a method so you can override a cache with a different configuration?